list.go

```diff
@@ -61,7 +63,7 @@ func (v *List) Unmarshal(buf []byte, rem int) ([]byte, int, error) {
 		return buf, rem, err
 	}
 	if len(v.Elems) == 0 {
-		return buf, rem, fmt.Errorf("cannot construct list with no elements")
+		v.T = typeNil{}
```

Comment:
This actually makes me realise there is a bug for the case when `len(v.Elems) > 0`. At the moment, the `v.T` would be left uninitialised, but it should be set to the type of `v.Elems[0]` and all elements should be checked to make sure they are consistent with that type.